### PR TITLE
Add support for arbitrary base URL to fix #9.

### LINKED ISSUE #9: Support installing releases from unofficial repos
Suggest adding `downloadBaseUrl` as an optional parameter.

### PR DESCRIPTION
**Pull request by brettle**
fix #9

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -41,6 +45,14 @@ runs:
         else
           webotsTag=${{ inputs.webotsVersion }}
         fi
+        if [ -n '${{ inputs.webotsBaseUrl }}' ]; then
+          if [ -n '${{ inputs.webotsTag }}' ]; then
+            echo "WARNING: Ignoring webotsTag=${{ inputs.webotsTag }} because webotsBaseUrl was specified."
+          fi
+          webotsBaseUrl=${{ inputs.webotsBaseUrl }}
+        else
+          webotsBaseUrl='https://github.com/cyberbotics/webots/releases/download/${{ inputs.webotsTag }}'
```
Comment (CoolSpy3) [inline action.yml]: Use the resolved tag to fix the CI
```suggestion
          webotsBaseUrl="https://github.com/cyberbotics/webots/releases/download/$webotsTag"
```

**Review by CoolSpy3 — CHANGES_REQUESTED** [review]

File: action.yml [hunk 2]
```diff
@@ -8,6 +8,10 @@ inputs:
     description: "Optionally specify a specific tag from the Webots repository to install. (Defaults to webotsVersion)"
     required: false
     default: ''
+  webotsBaseUrl:
+    description: "Optionally specify a base URL containing the distribution files (Defaults to https://github.com/cyberbotics/webots/releases/download/<webotsTag>)"
+    required: false
+    default: ''
```
Comment (CoolSpy3) [inline action.yml:14]:
```suggestion
    default: 'https://github.com/cyberbotics/webots/releases/download/'
```

File: action.yml [hunk 3]
```diff
@@ -73,7 +85,7 @@ runs:
         echo "binDir=${webotsBinDir}" >> $GITHUB_OUTPUT
         echo "bin=${webotsBin}" >> $GITHUB_OUTPUT
         echo "pkgName=$pkgName" >> $GITHUB_OUTPUT
-        echo "tag=$webotsTag" >> $GITHUB_OUTPUT
+        echo "baseUrl=$webotsBaseUrl" >> $GITHUB_OUTPUT
```
Comment (CoolSpy3) [inline action.yml:88]: The `tag` action output should be updated.

File: action.yml (hunk 2 above)
Comment (brettle) [reply action.yml:14]: I think we want that to remain empty so that the code detects it's absence. Similar to webotsTag.

File: action.yml (hunk 1 above)
Comment (brettle) [reply action.yml]: Fixed.

File: action.yml (hunk 3 above)
Comment (brettle) [reply action.yml:88]: I've replaced the tag output with baseUrl in my latest commit.

File: action.yml (hunk 2 above)
Comment (CoolSpy3) [reply action.yml:14]: Ah, my bad. I read the docs wrong.

**Review by CoolSpy3 — APPROVED** [review]

**Review by CoolSpy3 — CHANGES_REQUESTED** [review]
One more note: `webotsBaseUrl` should be passed to the action in the CI.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -27,6 +37,7 @@ jobs:
       with:
         webotsVersion: ${{ matrix.webotsVersion }}
         webotsTag: ${{ matrix.webotsTag || '' }}
+        webotsUrl: ${{ matrix.webotsUrl || '' }}
```
Comment (CoolSpy3) [inline .github/workflows/ci.yml]:
```suggestion
        webotsBaseUrl: ${{ matrix.webotsBaseUrl || '' }}
```

**Review by CoolSpy3 — CHANGES_REQUESTED** [review]

**Review by CoolSpy3 — APPROVED** [review]

**Pull request merged**